src/mbgl/annotation/annotation_manager.cpp

```diff
@@ -28,14 +37,17 @@ AnnotationManager::AnnotationManager(Style& style_)
 AnnotationManager::~AnnotationManager() = default;
 
 void AnnotationManager::setStyle(Style& style_) {
+    CHECK_ANNOTATIONS_ENABLED_AND_RETURN();
     style = style_;
 }
 
 void AnnotationManager::onStyleLoaded() {
+    CHECK_ANNOTATIONS_ENABLED_AND_RETURN();
     updateStyle();
 }
 
 AnnotationID AnnotationManager::addAnnotation(const Annotation& annotation) {
+    CHECK_ANNOTATIONS_ENABLED_AND_RETURN(nextID++);
```

Comment:
looks like `nextID` will be incremented twice when annotations are enabled.

Reply:
that's a macro call, so pre-processor will unfold it to:
```
if (!LayerManager::annotationsEnabled) {
    assert(false);                                   
    return nextID++;                                   
}
```
before compilation, thus `nextID` wont be incremented twice.